torch/distributed/device_mesh.py

```diff
@@ -553,7 +553,10 @@ def init_device_mesh(
                     f"Found len(mesh_dim_names): {len(mesh_dim_names)} and len(mesh_shape):{len(mesh_shape)}.",
                 )
 
-        mesh = torch.arange(math.prod(mesh_shape), dtype=torch.int).view(mesh_shape)
+        # Always initialize the mesh's tensor on CPU, regardless of what the
+        # external device type has been set to be (e.g. meta)
+        with torch.device("cpu"):
```

Comment:
Oh I think with this, we can remove the .detach().to("cpu") here: 
https://github.com/pytorch/pytorch/blob/main/torch/distributed/device_mesh.py#L215

Reply:
I was worrying that someone out there might init a `DeviceMesh` object directly instead of through `init_device_mesh`, so leaving that there might be safer.

Reply:
Just remember we had a check right above it, but ye I think we can keep it to make sure the dtype is consistent. 
https://github.com/pytorch/pytorch/blob/main/torch/distributed/device_mesh.py#L212-L213